### DISCUSSION TITLE
Controller + Grommet Select

### DISCUSSION THREAD

**Discussion by jonka5** [react-hook-form/react-hook-form#3668] [Q&A] [answered]
I have a Select from an external UI library wrapped in the react-hook-form Controller. My problem was that after an onChange the value was not updated immediately.

codesandbox: https://codesandbox.io/s/grommethook-form-vd5by

I solved this by executing the function from OnChange twice. Once with onBlur and the second time with onChange.

codesandbox: https://codesandbox.io/s/grommet-react-hook-form-forked-hrz0q?file=/src/App.js:884-922

The result works but I don't know if this is good pattern or if there are better ways.

**Comment by bluebill1049** [comment, marked as answer]
i would raise an issue at the other component `e.target.value` didn't supply any value, but this works: https://codesandbox.io/s/grommet-react-hook-form-forked-9fw0b?file=/src/App.js

**Reply by jonka5** [reply]
Thanks so much for the quick response. It helped a lot!

**Reply by bluebill1049** [reply]
np
